src/NUnitConsole/nunit3-console/TestEventHandler.cs

```diff
@@ -129,6 +138,8 @@ private void TestOutput(XmlNode outputNode)
             var testName = outputNode.GetAttribute("testname");
             var stream = outputNode.GetAttribute("stream");
 
+            FlushNewLineIfNecessary(testName);
+
```

Comment:
In this case, it seems like the right place. Because the new test is calling TestOutput we know it is about to write.